core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -285,6 +287,20 @@ trait WorkflowComponent {
       groupedSeq.map(_.toMap)
     }
 
+    def countWorkflowsByQueueStatusByUser: ReadAction[Map[String, Map[String, Int]]] = {
+      // Query for a Seq of (User, Workflow) tuples.
+      val userWorkflows = (for {
+        workflow <- findQueuedAndRunningWorkflows
+        submission <- submissionQuery if workflow.submissionId === submission.id
+        user <- rawlsUserQuery if submission.submitterId === user.userSubjectId
+      } yield (user, workflow)).result
+
+      // Combine results into a nested map using Scalaz semigroups for aggregation.
```

Comment:
Thanks for the explanation.  There are definitely other places in Rawls where we do this, and it might be good to convert them to semigroups once we're more comfortable.  But for now, I would like a comment along the lines of "this operator combines maps' keys and sums their values"

A general overview of these concepts would be excellent.  We're pretty comfortable with monads (or at least map/flatMap) but not the rest of the family.  Red Team (Cromwell) does more Scalaz.  As we grow closer organizationally maybe they will have more of an influence on us...